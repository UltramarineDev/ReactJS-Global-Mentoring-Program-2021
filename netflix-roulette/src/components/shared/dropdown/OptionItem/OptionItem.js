import React from 'react';
import PropTypes from 'prop-types';

import styles from './OptionItem.module.scss';

const OptionItem = ({ label, onClick, value }) => {
  const onLinkClick = () => onClick(value);

  return (
    <a aria-label="option_item" onClick={onLinkClick} className={styles.item}>{label}</a>
  );
};

OptionItem.propTypes = {
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  value: PropTypes.object.isRequired,
};

export default OptionItem;
