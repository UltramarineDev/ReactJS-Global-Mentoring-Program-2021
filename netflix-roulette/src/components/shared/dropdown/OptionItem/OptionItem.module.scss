.item {
  display: block;
  padding: 12px 12px;
}

.item:hover {
  background-color: #f65261;
}
