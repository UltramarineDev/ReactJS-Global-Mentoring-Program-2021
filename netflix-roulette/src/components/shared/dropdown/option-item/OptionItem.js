import React from 'react';
import PropTypes from 'prop-types';

import styles from './OptionItem.module.scss';

const OptionItemComponent = ({ label, onClick }) => (
  <>
    <div className={styles.item}>
      <a onClick={onClick}>{label}</a>
    </div>
  </>
);

OptionItemComponent.propTypes = {
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default OptionItemComponent;
