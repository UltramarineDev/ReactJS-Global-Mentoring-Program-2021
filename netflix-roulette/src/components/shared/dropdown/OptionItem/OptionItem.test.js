import React from 'react';
import { render, cleanup } from '@testing-library/react';

import OptionItem from './OptionItem';

describe('OptionItem tests', () => {
  afterEach(() => {
    cleanup();
  });
  it('renders OptionItem', () => {
    const label = '';
    const value = {};
    const onClick = jest.fn();
    const { asFragment } = render(<OptionItem label={label} onClick={onClick} value={value} />);
    expect(asFragment(<OptionItem label={label} onClick={onClick} value={value} />)).toMatchSnapshot();
  });
});
