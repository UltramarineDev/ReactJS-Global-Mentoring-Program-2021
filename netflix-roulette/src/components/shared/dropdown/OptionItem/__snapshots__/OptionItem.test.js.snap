// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionItem tests renders OptionItem 1`] = `
<DocumentFragment>
  <a
    aria-label="option_item"
    class="item"
  />
</DocumentFragment>
`;
